src/main/scala/chisel3/util/MixedVec.scala

```diff
@@ -0,0 +1,103 @@
+// See LICENSE for license details.
+
+package chisel3.util
+
+import chisel3._
+import chisel3.core.{Data, requireIsChiselType, requireIsHardware}
+import chisel3.internal.naming.chiselName
+
+import scala.collection.immutable.ListMap
+
+object MixedVecWireInit {
```

Comment:
This seems like a lot of words. Why not MixedVecInit? Note that VecInit also internally creates a wire.

Also, thoughts on a varargs version vs. the Seq version? In the comments for VecInit, it appears that we would have preferred the varargs version for consistency with Scala.

Reply:
1) Mainly to make room for the introduction of Vec (and associated MixedVec) literals (#849) if it ever happens.

2) I wouldn't be opposed to a varargs version in addition to a Seq version, though I would prefer it if the Seq version were not completely removed, since sometimes we do want to programmatically create a Seq to use as the type and not have to worry about `:_*` etc.

Reply:
1. I think that whatever we do, we at least need to keep Vec and MixedVec consistent, and ideally would also keep them consistent with their closest Scala analogues. I'm not sure how we want to do Vec literals in the future, though we might want to keep that consistent with Bundle literals (so Vec(...).Lit, or even Vec.Lit). Or maybe VecLit. But the alternative solution here would be to deprecate VecInit and have it be VecWireInit (or just VecWire?), but we'd end up waiting at least one deprecation cycle (if ever) before we could introduce a different meaning  for VecInit. So I highly doubt that's a practical pathway.
2. The argument for a varargs version is that it's more consistent with the Scala collections library, where you do Seq(elems*), or elems.toSeq. So if we didn't already have VecInit(Seq), a consistent solution might be mySeq.toVec (or mySeq.toVecWire). The idea behind consistency is so that programmers can move between Scala and Chisel without having to think about a different set of interfaces.

Reply:
1) I'm happy to rename to `MixedVecInit` and will do so tomorrow.

2) So in conclusion can we keep the Seq version as well, since we don't have `.toVec` or things like that?

Reply:
I think it's worth a discussion about whether we want .toVec (or similar), since I think there can be issues with a constructor that takes both a varargs and a Seq.

Reply:
I agree -  since (I think) `.toVec` would be a further API addition, perhaps we can merge this PR first and propose/discuss `.toVec` in a future issue/PR?